File: Cloudella 1.0 C# trial/Cloudella/Node.cs
using System.Text.Json;

namespace DynamicNode;

public class Node
{
    public string name { get; set; }
    public string text { get; set; }
    public dynamic value { get; set; }

    public Node()
    {
    }

    public Node(Node node)
    {
        name = node.name;
        text = node.text;
        value = node.value;
    }

    public static Node? ToNode(string json)
    {
        return JsonSerializer.Deserialize<Node>(json);
    }

    public static string ToJson(Node node)
    {
        return JsonSerializer.Serialize(node);
    }

    public async void SaveNodeToFileAsync(string path)
    {
        string title = "/" + this.name + ".node.json";
        string json = Node.ToJson(this);

        await File.WriteAllTextAsync(path + title, json);
    }
}

File: Cloudella 1.1.1/Node.cpp
#include "Node.h"

json Node::toJSON() const
{
    json jNode{};
    jNode["name"] = this->name;
    jNode["text"] = this->text;
    return jNode;
}

void Node::addNodeToDir(const std::string& path) const
{
    std::string filename(this->name + ".node" + ".json");
    std::string filepath = (path + "/" + filename);

    std::fstream output_fstream;
    output_fstream.open(filepath, std::ios_base::out);

    json tempJSON = this->toJSON();
    output_fstream << tempJSON << '\n';
}

